flang/lib/Lower/PFTBuilder.cpp

```diff
@@ -1715,3 +1715,46 @@ Fortran::lower::pft::buildFuncResultDependencyList(
   variableList[0].pop_back();
   return variableList[0];
 }
+
+namespace {
+/// Helper class to find all the symbols referenced in a FunctionLikeUnit.
+/// It defines a parse tree visitor doing a deep visit in all nodes with
+/// symbols (including evaluate::Expr).
+struct SymbolVisitor {
+  template <typename A>
+  bool Pre(const A &x) {
+    if constexpr (Fortran::parser::HasTypedExpr<A>::value)
+      if (const auto *expr = Fortran::semantics::GetExpr(x))
+        visitExpr(*expr);
+    return true;
+  };
```

Comment:
I get a warning (and thus a build error) because this line has an unnecessary ";".

I'm using GNU 9.3.0 for my builds.